### PR TITLE
Add test for URL issue 348

### PR DESCRIPTION
**Pull request by achristensen07**

### CODE REVIEW

**Review by annevk — APPROVED** [review]
Thanks for writing tests too! This looks good to me, but I'll let @domenic verify per usual.

**Comment by annevk** [comment]
Decided to land this anyway given that @achristensen07 did all the work required and it's a trivial enough change.

File: url/resources/urltestdata.json [hunk 1]
```diff
@@ -4015,6 +4015,37 @@
...
+    "port": "",
+    "pathname": "/",
+    "search": "?%27",
+    "hash": ""
+  },
+  {
+    "input": "notspecial://host/?'",
+    "base": "about:blank",
+    "href": "notspecial://host/?'",
+    "origin": "notspecial://host",
```
Comment (domenic) [inline url/resources/urltestdata.json]: origin should be "null"

**Review by domenic — COMMENTED** [review]
This isn't quite right.

**Comment by annevk** [comment]
Sigh, I should learn to wait.

**Comment by annevk** [comment]
I'll post a fix.

**Pull request merged**